### ISSUE TITLE
Regular failure on mpeg-dash video seeking on http2

### ISSUE THREAD

**Issue by nimatrueway** [mitmproxy/mitmproxy#2823]
Hey guys. Thanks for this incredible tool. I tried to write a script that manipulates SNI of TLS connections in all requests and got no problem regarding that.

However when I load a **page with video (MPEG-DASH)** say a facebook video, the connection **quality** goes **very bad**. It takes a **long time** if it doesn't fail when you want to **seek** back and forth through a video. Can anybody explain why this happens ?

#### 1. Run mitmdump
```
mitmdump --listen-port 9990 --ssl-insecure
```

##### 4. Open a video on facebook [like this](https://www.facebook.com/safarilive/videos/1210725652404731/) and try to seek randomly.

**Comment by mhils** [comment]
Have you tried enabling HTTP response streaming in mitmproxy? This can be done by setting the `stream_large_bodies` in mitmproxy 3.0 or by using the corresponding command line flag in 2.0.

**Comment by nimatrueway** [comment]
Yea @mhils, tried all of these but couldn't fix it.
1. `--set stream_large_bodies=0`
2. `--set stream_large_bodies=1`
3. `--set stream_large_bodies=250k`

**Comment by mhils** [comment]
1. Does it work without your script?
2. Can you view the response in mitmproxy? 
3. Are we on HTTP/1 or HTTP/2?

**Comment by nimatrueway** [comment]
1. Interesting question ! No, even with no script mitmproxy can't relay it properly.
2. There is no response to those requests aiming video data. 🤔 
3. HTTP/2 

I tried a couple of other websites and figured HTTP/1.1 is OK and the problem is within HTTP/2 responses.

**Comment by mhils** [comment]
Just to be sure, does it work if you force HTTP/1 (`--set http2=false`) with all other things being equal?

**Comment by nimatrueway** [comment]
Problem went after forcing HTTP/1 👍

**Comment by mhils** [comment]
@Kriechi, any ideas?

**Comment by Kriechi** [comment]
I'm not familiar with MPEG streaming.

* There could be some latency / round-trip measurement that gets screwed up between client<->proxy and proxy<->server.
* What is actually transferred? Is it a single HTTP request+response, or does MPEG-dash make use of some fancy H2 feature (pushed streams? windowing?)?

Would be good to have a test site. @nimatrueway can you provide a link, or can we build a minimalistic MPEG-DASH example + index.html?

**Comment by nimatrueway** [comment]
It's a really weird (performance/load kind of problem) problem guys. Today I spent hours trying to find out what is it exactly that goes wrong so that I can elaborate here but I couldn't.

Let me close this issue for the moment, because this project is really hot and is already very busy with more important issues. I'll re-open it whenever I got solid evidence and could spot the exact bug.

Thank you for your immediate response and care.
